.github/workflows/build.yml

```diff
@@ -0,0 +1,123 @@
+name: Build and upload to PyPI
+# publish when a (published) GitHub Release is created
+on:
+  push:
+  pull_request:
+  release:
+    types:
+      - published
+
+jobs:
+  build_wheels:
+    name: Build wheels [${{ matrix.os }}, ${{ matrix.platform }}, py ${{ matrix.python-version }}]
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        platform: [x64]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Set up python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install cibuildwheel
+        run: |
+          python -m pip install cibuildwheel==1.6.3
+      - name: Install Visual C++ for Python 2.7
+        if: runner.os == 'Windows'
+        run: |
+          choco install vcpython27 -f -y
+
+      - name: Build wheels on ${{ matrix.os}}
+        run: |
+          RUNNER_OS="${{ runner.os }}"
+          PLATFORM="${{ matrix.platform }}"
+          PYTHON_VERSION="${{ matrix.python-version }}"
+          echo $RUNNER_OS
+          echo $PLATFORM
+          echo $PYTHON_VERSION
+
+          if [ "$RUNNER_OS" == "Windows" ]; then
+            if [ "$PLATFORM" == "x64" ]; then
+              if [ "$PYTHON_VERSION" == "3.6" ]; then
+                export CIBW_BUILD="cp36-win_amd64"
+              elif [ "$PYTHON_VERSION" == "3.7" ]; then
+                export CIBW_BUILD="cp37-win_amd64"
+              elif [ "$PYTHON_VERSION" == "3.8" ]; then
+                export CIBW_BUILD="cp38-win_amd64"  
+              fi
+            elif [ "$PLATFORM" == "x32" ]; then
+              export CIBW_BUILD="cp36-win32 cp37-win32 cp38-win32"
```

Comment:
Can't you do this for the other platforms too? Why run each build seperately?

Reply:
Two things, when I tried to run all the builds together (through `CIBW_BUILD`), I ran into issues. Now that I've learned a bit (like that multiple python version can be one wheel, etc). I might can condense this down. I don't think there is a way around running off of the proper machine though, i.e. Windows builds must be run from a windows machine. This is a little counter-intuitive since its running on docker, but I found a note about it somewhere. 

The other thing I think you bring up is the lack of x32 for the other platforms? I started to build that out, but then we decided to only build x64 for now so I stopped but left what I had done in there. Would you rather I remove it?

Reply:
I'll take another look at trying to run these together now that I understand `cibuildwheel` a bit better.

Reply:
> Windows builds must be run from a windows machine

right, we will need seperate windows, macosx and linux builds

Reply:
fixed this up and made it much cleaner